### PR TITLE
Develop -> Master

### PR DESCRIPTION
**Pull request by samgomena**
This will break master but is required to get the changes on my branch (`sgomena/develop`) into `develop` after a faulty PR merge that happened earlier.

Just for clarification, please do not merge this without:
1. Approving the PR
2. Contactinting or confirming the merge with the _author_ of the PR.

### CODE REVIEW

File: test/features/application.feature [hunk 1]
```diff
@@ -3,17 +3,36 @@
...
+
+  Scenario: Verify can logout and login again
+    When I spin up the application
+      And I open the login window
+      And I login in to the application
+      And I open the application window
+      And I logout of the application
+    Then the application window should be hidden
+    When I wait 2 seconds
+      # TODO: Have to 'manually' define the `login_window` object here; it should be done automagically
```
Comment (payaljoshi0189) [inline test/features/application.feature:36]: Just a spell change from 'automagically' to 'automatically'

File: test/steps/step_login.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@when(r'I open the login window')
+def open_login_window(context):
+    context.login_window = context.__app.login
+
+
+@when(r'I login in to the application')
+def login(context):
+    assert_is_not(context.login_window, None)
+
+    QTest.keyClicks(context.login_window.login.username, "samgomena@gmail.com")
```
Comment (payaljoshi0189) [inline test/steps/step_login.py:21]: A suggestion, could we remove hardcoded value(samgomena@gmail.com) and have it read the value using a variable?

**Review by payaljoshi0189 — CHANGES_REQUESTED** [review]
I have left a few comments before this is merged.Overall, this looks good.

**Comment by samgomena** [comment]
Fixes for your comments are coming in across a couple other PR's that have/will be created. Going to close this and re-open once `develop` updated.

**Pull request closed without merging**